README.md

```diff
@@ -62,28 +62,98 @@ import (
 	"github.com/marianogappa/parseq"
 )
 
+type DataMapper struct {
+	// you can put private fields here
+}
+
+func (p *DataMapper) Map(input int) int {
+	// access go routine-private data
+	time.Sleep(time.Duration(200) * time.Millisecond) // processing a request takes 1s
+	// process input value, and return result
+	return input
+}
+
 func main() {
-	p := parseq.New(5, process)			// 5 goroutines using the process function
+	p, err := parseq.NewWithMapper[int, int](5, &DataMapper{}) // 5 goroutines using the process function
+	if err != nil {
+		panic(err)
+	}
 
 	go p.Start()
 	go makeRequests(p)
 
-	for out := range p.Output {			// after initial 1s, requests output every ~200ms
-		fmt.Print(out.(int), ".")		// and output respects input order
+	for out := range p.Output { // after initial 1s, requests output every ~200ms
+		fmt.Print(out, ".") // and output respects input order
 	}
+	p.Close()
 }
 
-func makeRequests(p parseq.ParSeq) {
+func makeRequests(p *parseq.ParSeq[int, int]) {
 	counter := 666
 	for {
-		p.Input <- counter			// this simulates an incoming request
-		time.Sleep(200 * time.Millisecond)	// requests come every 200ms
+		p.Input <- counter                 // this simulates an incoming request
+		time.Sleep(200 * time.Millisecond) // requests come every 200ms
 		counter++
 	}
 }
+```
+
+Sometime it is necessary to keep per-thread/go-routine state, or other data critical for mapping the input data, that must not be shared between threads. In this case, instead of providing a single mapper instance, you can provide a slice of mappers. The size of the slice must match the `parallelism` used when creating the `parseq` instance.
+
+```go
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/marianogappa/parseq"
+)
+
+type DataMapper struct {
+	// You can put private fields here.
+	// For demonstative purposes we'll use this to
+	// set an artificial delay used in the Map func.
+	delay int
+}
+
+func (p *DataMapper) Map(input int) int {
+	// access go routine-private data
+	time.Sleep(time.Duration(p.delay) * time.Millisecond) // processing a request takes 1s
```

Comment:
Comment looks wrong. Doesn't take 1s.

Reply:
i improved the whole code block instead of just fixing that comment. Please have a look

Reply:
Looks good.